### PR TITLE
Always fake ProcessManager when you fake Filesystem in tests

### PR DESCRIPTION
**Pull request by Hixie**
...because otherwise, processes that think they're manipulating your
filesystem will be doing crazy things the test is ignoring, leading to
(at best) failures and (at worst) flakes or disk corruption.

### CODE REVIEW

**Comment by Hixie** [comment]
cc @jonahwilliams this fixes the ThrowingPub layering issue too

File: packages/flutter_tools/test/src/fake_process_manager.dart [hunk 1]
```diff
@@ -0,0 +1,206 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
```
Comment (zanderso) [inline packages/flutter_tools/test/src/fake_process_manager.dart]: This file needs comments.

Comment (Hixie) [reply packages/flutter_tools/test/src/fake_process_manager.dart]: done

Comment (zanderso) [reply packages/flutter_tools/test/src/fake_process_manager.dart]: Cool, thanks. This is a really nice API, but it doesn't appear to have any non-trivial uses in this PR. What are your plans for it?

Comment (Hixie) [reply packages/flutter_tools/test/src/fake_process_manager.dart]: My first customer is https://github.com/flutter/flutter/pull/42491. I suspect the API will evolve as it gets more users.

**Review by zanderso — APPROVED** [review]
lgtm

**Pull request merged**